docs/streams/developer-guide/dsl-topology-naming.html

```diff
@@ -0,0 +1,290 @@
+		  (also now with <code>StreamJoined</code>)
+		  <code>Grouped</code> classes and state store (and changelog topics) with <code>Materialized</code>.  
+		  But it's worth reiterating the importance of naming these DSL topology operations again.
+
+		  Here's how your DSL code looks now giving a specific name to your state store:
+		  <code>
+		    builder.&gt;String, String&lt;stream("input")
+		            .filter((k, v) -> v != null && v.length() >= 6)
+		            .groupByKey()
+		            .count(Materialized.as("Purchase_count_store"))
+		            .toStream()
+		            .to("output");
+		  </code>
+
+		  And here's the topology 
+
+		  <code>
+		  Topologies:
+		   Sub-topology: 0
+		    Source: KSTREAM-SOURCE-0000000000 (topics: [input])
+		      --> KSTREAM-FILTER-0000000001
+		    Processor: KSTREAM-FILTER-0000000001 (stores: [])
+		      --> KSTREAM-AGGREGATE-0000000002
+		      <-- KSTREAM-SOURCE-0000000000
+		    Processor: KSTREAM-AGGREGATE-0000000002 (stores: [Purchase_count_store])
+		      --> KTABLE-TOSTREAM-0000000003
+		      <-- KSTREAM-FILTER-0000000001
+		    Processor: KTABLE-TOSTREAM-0000000003 (stores: [])
+		      --> KSTREAM-SINK-0000000004
+		      <-- KSTREAM-AGGREGATE-0000000002
+		    Sink: KSTREAM-SINK-0000000004 (topic: output)
+		      <-- KTABLE-TOSTREAM-0000000003
+		  </code>
+
+		  So now, even though you've added processors before your state store, the store name and its changelog 
+		  topic names don't change!  This makes your topology more robust and resilient to changes made by 
+		  adding or removing processors.
+
+		<h2>Conclusion</h2>
+
+		 So to summarize, it's a good practice to name your processing nodes when using the DSL, and it's even
+		 more important to do so when you have processing that remains when re-start your application - repartition
+		 topics and state stores (and the accompanying changelog topics).
+
+		 A couple of points we should go over again regarding naming your DSL topology:
+		<ol>
+			<li>
+				If you have an <em>existing topology</em> and you've <em>haven't</em> named your
+				state stores (hence chagelog topics) and repartition topics it's recommended that you
+				do so.  But this will be a topology breaking change, so you'll need to shut down all
+				application instances, make the changes then run the
+				<a href="/{{version}}/documentation/streams/developer-guide/app-reset-tool">Streams Reset Tool</a>.
+				Although this may be inconvenient at first, it's worth the effort to protect your application from
+				unexpected errors due to topology changes.
+			</li>
+			<li>
+				If you have a <em>new topology</em> make sure you name the persistent parts of your topology
+				: state stores (changelog topics) and repartition topics.  That way once you deploy your
+				application you are protected from topology changes breaking your Kafka Streams application.
+				Even if you don't want to add names to transient processors at first, that's fine as you can
```

Comment:
```suggestion
				If you don't want to add names to transient processors at first, that's fine, because you can
```

Reply:
nit: `transient` --> `stateless` ? I think it's good to be consistent in our terminology throughout the docs -- it might be obvious to us that "stateless", "in-memory", and "transient" all refer to the same thing, but I think users will find this confusing.

Reply:
good point